Virtualization docs fixes during ROSA review (#69051)
openshift/openshift-docs: modules/virt-configuring-storage-class-bootsource-update.adoc

      template:
        spec:
          storageClassName: <new_storage_class> <1>

>       schedule: "0 */12 * * *" <2>

Reviewer: this section is about changing the storage class for boot images. how come we suddenly want to include the schedule/managedDataSource field?

Author: @akalenyu  If I am reading this right, in order to add `spec.dataImportCronTemplates.spec.template.spec.storageClassName` parameter, we need to add the `schedule` and `managedDataSource` parameters. Am I doing something wrong? 

```
# Please edit the object below. Lines beginning with a '#' will be ignored,
# and an empty file will abort the edit. If an error occurs while saving this file will be
# reopened with the relevant failures.
#
# hyperconvergeds.hco.kubevirt.io "kubevirt-hyperconverged" was not valid:
# * spec.dataImportCronTemplates[0].spec.managedDataSource: Required value
# * spec.dataImportCronTemplates[0].spec.schedule: Required value
```

Reviewer: ah you're right, this was an actual doc bug then

      managedDataSource: <data_source> <3>
# ...
----
<1> Define the storage class.
<2> Required: Schedule for the job specified in cron format.

> <3>  Required: For the custom image to be detected as an available boot source, the value of the `managedDataSource` field must match the value of the `spec.dataVolumeTemplates.spec.sourceRef.name` field in the VM template YAML file.

Author: @awels Thanks for your feedback. I might have gotten the wording from elsewhere in the docs, [Enabling automatic updates for custom boot sources, callout 4](https://docs.openshift.com/container-platform/4.15/virt/storage/virt-automatic-bootsource-updates.html#virt-autoupdate-custom-bootsource_virt-automatic-bootsource-updates). 

> For the custom image to be detected as an available boot source, the name of the image's `managedDataSource` must match the name of the template's `DataSource`, which is found under `spec.dataVolumeTemplates.spec.sourceRef.name` in the VM template YAML file.

> <3>  Required: For the custom image to be detected as an available boot source, the value of the `managedDataSource` field must match the value of the `spec.dataVolumeTemplates.spec.sourceRef.name` field in the VM template YAML file.

Reviewer 2: Like I said it is technically correct, as the two need to match, but it becomes incorrect when you have multiple VM templates referencing the same dataSourceTemplate. The VM template must match the DS template, not the other way around. And the way it is worded right now it seems like the DS template must match the VM template, which is correct most of the time but in the one case I mention it might not be correct.

> <3>  Required: For the custom image to be detected as an available boot source, the value of the `managedDataSource` field must match the value of the `spec.dataVolumeTemplates.spec.sourceRef.name` field in the VM template YAML file.

Author: Ugh. Sorry this is all fairly new to me. [One more attempt](https://github.com/openshift/openshift-docs/pull/69051/commits/e136dc4b7ef059e8f196027edfadf7a1ab3005a4)! 

> For the custom image to be detected as an available boot source, the value of the `spec.dataVolumeTemplates.spec.sourceRef.name` parameter in the VM template must this value.

> <3> Required: For the custom image to be detected as an available boot source, the name of the image's `managedDataSource` must match the name of the template's `DataSource`, which is found under `spec.dataVolumeTemplates.spec.sourceRef.name` in the VM template YAML file.

Author: **QE REVIEW NEEDED** Please verify that these parameters are required and the definitions are correct.

> <3> Required: For the custom image to be detected as an available boot source, the name of the image's `managedDataSource` must match the name of the template's `DataSource`, which is found under `spec.dataVolumeTemplates.spec.sourceRef.name` in the VM template YAML file.

Reviewer 3: Suggestion to potentially improve clarity:
```suggestion
<3> Required: For the custom image to be detected as an available boot source, the value of the `managedDataSource` field must match the value of the `spec.dataVolumeTemplates.spec.sourceRef.name` field in the VM template YAML file.
```

----
$ oc patch storageclass <new_storage_class> -p '{"metadata":{"annotations":{"storageclass.kubernetes.io/is-default-class":"true"}}}' <1>
----

> <1> Replace `<new_storage_class>` with the `storageClassName` value that you added to the `HyperConverged` CR.

Reviewer 4: I think we need to edit some things at the beginning of the doc as well.

`Specify a new default storage class in the `HyperConverged` custom resource (CR).` - does not sound right. We need either a default storage class, or a specified in the HyperConverged CR, which should override the default.

In addition to that, since 4.15 we have a default-virt annotation
`storageclass.kubevirt.io/is-default-virt-class: "true"`
that will be preferred over the kubernetes default annotation
`storageclass.kubernetes.io/is-default-class":"true"`

This is a bit information about this annotation: https://github.com/kubevirt/community/blob/main/design-proposals/default-virt-storage-class.md
(please pay attention, in the design proposal there's an outdated annotation (different wording), but here I pasted the correct one)

Author: @ousleyp Are you awar of this? It seem like a matter that should be addressed on its own. 

> In addition to that, since 4.15 we have a default-virt annotation
> storageclass.kubevirt.io/is-default-virt-class: "true"
> that will be preferred over the kubernetes default annotation
> storageclass.kubernetes.io/is-default-class":"true"

Reviewer: virt sc was handled in https://github.com/openshift/openshift-docs/pull/70639

Reviewer 4: Right, it was addressed separately, but if we have a section about the default storage class for the boot source images - we should mention here both virt default and kubernetes default.

virt default will be preferred, but it will not show the **(default)** part, when we do `oc get storageclass`

Reviewer 4: @mburke5678 so will we address this as a separate issue?
